packages/headless/src/controllers/controller/error-payload.ts

```diff
@@ -1,16 +1,3 @@
-export interface ErrorPayload {
-  /**
-   * The HTTP status code.
-   */
-  statusCode: number;
+import {SearchAPIErrorWithStatusCode} from '../../api/search/search-api-error-response';
```

Comment:
@samisayegh this interface was a bit of a duplicate, and SearchAPIErrorWithStatusCode was used in the search state, so I referenced it directly. Perhaps we could just use SearchAPIErrorWithStatusCode everywhere?

Reply:
The issue is an interface must be extracted to retrieve it's annotations. However, I don't think it's good to export an interface with `SearchAPI` in it's name. The alternative is to rename the internal interface, but this makes the core more abstract.

The middle-ground was to create a new interface that was exported and place the documentation on it. The core interfaces map to the exported interface.

See this thread for the original discussion: https://github.com/coveo/ui-kit/pull/541#discussion_r582368566

Reply:
ahh got it